dev/benchmark: add sysbench in k8s benchmark report

cofyc opened the pull request:
<!--Thanks for your contribution to TiDB documentation. See [CONTRIBUTING](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md) before filing this PR.-->

### What is changed, added or deleted? <!--Required-->

Add TiDB sysbench benchmark report in k8s (gke).

### Which version does your change affect? <!--Required; write "N/A" if it is not applicable-->

dev

https://github.com/cofyc/docs-cn/blob/sysbench-in-k8s/dev/benchmark/sysbench-in-k8s.md

weekface commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+[server]
+status-addr = "0.0.0.0:20180"
+grpc-concurrency = 6
+[readpool.storage]
```
block-cache 在 tidb 3.0 的配置文件格式不是这个了吧？

cofyc replied:
是的，不过这些配置在 3.0 继续兼容。

weekface commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| TiDB     | c2-standard-16 | 3     |
+| Sysbench | c2-standard-30 | 1     |
+
+在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 没有 region 有 3 个可用区同时提供 c2 机器，我们选择了如何机器型号进行测试:
```
```suggestion
在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 没有 region 有 3 个可用区同时提供 c2 机器，我们选择了如下机器型号进行测试:
```

weekface commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+![Service vs Load Balancer](/media/sysbench-in-k8s/service-vs-load-balancer-latency.svg)
+
+从图中可以看到 Ubuntu 系统在单纯的 Point Select 测试是中表现比 COS 系统要好。
```
```suggestion
从图中可以看到 K8S Service 在单纯的 Point Select 测试是中表现比 GCP LoadBalancer 要好。
```

weekface commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+- 使用 Load Balancer 在集群外访问，会略损失性能 (~5%)
+- 多可用区下节点之间延迟增加，会对 TiDB 性能产生一定的影响 (30% ~ 6%，随并发数增加而下降)
+
+但要注意的是，这些因素可能随着时间变化，不同公有云下的表现可能会略有不同。同时，sysbench 测试用例并不能代表真实应用场景，在做选择前需做实际生产测试，并综合不通选择背后的代价（操作系统差异、Host 网络的限制等）。
```
“并综合不通选择背后的代价”？

cofyc replied:
fixed

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+## 目的
+
+- 测试典型公有云平台上 TiDB 性能数据
+- 测试公有云平台磁盘、网络、CPU 以及不同 Pod 网络下 对 TiDB 性能影响
```
```suggestion
- 测试公有云平台磁盘、网络、CPU 以及不同 Pod 网络下对 TiDB 性能的影响
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+TiDB Operator 使用 v1.0.0 版本。
+
+PD, TiDB 和 TiKV 实例数均为 3 个。各组件分别作了如下配置，未配置部分使用默认值。
```
```suggestion
PD、TiDB 和 TiKV 实例数均为 3 个。各组件分别作了如下配置，未配置部分使用默认值。
```

lilin90 replied:
这里的“实例”指的是 Pod 吗？

cofyc replied:
是的

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| TiDB     | c2-standard-16 | 3     |
+| Sysbench | c2-standard-30 | 1     |
+
+在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 没有 region 有 3 个可用区同时提供 c2 机器，我们选择了如下机器型号进行测试:
```
```suggestion
在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 的 Region 均没有 3 个可用区可同时提供 c2 机器，我们选择了如下机器型号进行测试：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+#### 磁盘
+
+GKE 当前的 NVMe 磁盘还在 Alpha 阶段，使用需特殊申请，不具备普遍意义。测试中，本地 SSD 盘统一使用 iSCSI 接口类型。挂载参数参考[官方建议](https://cloud.google.com/compute/docs/disks/performance#optimize_local_ssd)增加了 `discard,nobarrier` 选项。完整挂载例子如下:
```
```suggestion
GKE 当前的 NVMe 磁盘还在 Alpha 阶段，使用需特殊申请，不具备普遍意义。测试中，本地 SSD 盘统一使用 iSCSI 接口类型。挂载参数参考[官方建议](https://cloud.google.com/compute/docs/disks/performance#optimize_local_ssd)增加了 `discard,nobarrier` 选项。完整挂载例子如下：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+#### CPU
+
+在单可用集群测试中，我们为 TiDB/TiKV 选择 c2-standard-16 机型测试。
+在单可用于多可用集群对比测试中，因为 GCP 没有区域可以同时在 3 可用区里申请
```
```suggestion
在单可用与多可用集群的对比测试中，因为 GCP 没有区域可以同时在 3 个可用区里申请
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+在单可用集群测试中，我们为 TiDB/TiKV 选择 c2-standard-16 机型测试。
+在单可用于多可用集群对比测试中，因为 GCP 没有区域可以同时在 3 可用区里申请
+c2-standard-16 机型。我们选择 n1-standard-16 机型测试。
```
```suggestion
c2-standard-16 机型。我们选择了 n1-standard-16 机型测试。
```

lilin90 replied:
Do not break lines manually unless you want to put them in different paras.

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+### 操作系统及参数
+
+GKE 支持两种操作系统：COS 和 Ubuntu 。在 Point Select 测试中进行了对比测试。其余测试中，同一使用 Ubuntu 系统进行测试。
```
Do you mean "CentOS" by COS? If yes, let's make it accurate.

```suggestion
GKE 支持两种操作系统：CentOS 和 Ubuntu 。在 Point Select 测试中进行了对比测试。其余测试中，统一使用 Ubuntu 系统进行测试。
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
````diff
@@ -0,0 +1,407 @@
+sysctl net.ipv4.tcp_syncookies=0
+```
+
+同时对最大文件数配置为 1000000 。
````
```suggestion
同时对最大文件数配置为 1000000。
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+### Sysbench 版本与运行参数
+
+本次测试中 sysbench 版本为 1.0.17 。并测试前统一使用 `oltp_common` 的 `prewarm` 命令进行数据预热。
```
```suggestion
本次测试中 sysbench 版本为 1.0.17。并在测试前统一使用 `oltp_common` 的 `prewarm` 命令进行数据预热。
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
````diff
@@ -0,0 +1,407 @@
+  prepare
+```
+
+`<tidb-host`> 为 tidb 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP, Service 域名，Host IP 以及 Load Balancer IP 。（下同）
````
```suggestion
`<tidb-host`> 为 TiDB 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP、Service 域名、Host IP 以及 Load Balancer IP。（下同）
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
````diff
@@ -0,0 +1,407 @@
+  run
+```
+
+`<test>` 为 sysbench 的测试 case 。我们选择了 oltp_point_select, oltp_update_index, oltp_update_no_index, oltp_read_write 这几种。
````
```suggestion
`<test>` 为 sysbench 的测试 case。我们选择了 oltp_point_select、oltp_update_index、oltp_update_no_index、oltp_read_write 这几种。
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| 1200    | 441084.13 | 5.28            |
+| 1500    | 447659.15 | 6.67            |
+
+QPS 对比:
```
```suggestion
QPS 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+![Pod vs Host Network](/media/sysbench-in-k8s/pod-vs-host-network-qps.svg)
+
+Latency 对比:
```
```suggestion
Latency 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+GKE 平台可以为节点选择 [Ubuntu 和 COS 两种操作系统](https://cloud.google.com/kubernetes-engine/docs/concepts/node-images)。本次测试中，分别对两种操作系统进行 Point Select 测试。
+
+此次测试中 Pod 网络模式为 Host 。
```
```suggestion
此次测试中 Pod 网络模式为 Host。
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| 1200    | 489220.93 | 4.33            |
+| 1500    | 489988.97 | 5.47            |
+
+QPS 对比:
```
```suggestion
QPS 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+![COS vs Ubuntu](/media/sysbench-in-k8s/cos-vs-ubuntu-qps.svg)
+
+Latency 对比:
```
```suggestion
Latency 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| 1200    | 441084.13 | 5.28            |
+| 1500    | 447659.15 | 6.67            |
+
+QPS 对比:
```
```suggestion
QPS 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+![Service vs Load Balancer](/media/sysbench-in-k8s/service-vs-load-balancer-qps.svg)
+
+Latency 对比:
```
```suggestion
Latency 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+GCP 多可用区，涉及跨 Zone 通信，网络延迟相比同 Zone 会少许增加。我们使用同样机器配置，对两种部署方案进行同一标准下的性能测试，了解多可用区延迟增加带来的影响。
+
+单可用区:
```
```suggestion
单可用区：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| 1200    | 294765.83 | 8.43            |
+| 1500    | 298619.31 | 10.27           |
+
+多可用区:
```
```suggestion
多可用区：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+| 1200    | 258835.24 | 10.09           |
+| 1500    | 280114.00 | 12.75           |
+
+QPS 对比:
```
```suggestion
QPS 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+![Single Zonal vs Regional](/media/sysbench-in-k8s/single-zonal-vs-regional-qps.svg)
+
+Latency 对比:
```
```suggestion
Latency 对比：
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+此次测试主要将典型公有云部署 Kubernetes 运行 TiDB 集群的几种场景使用 sysbench 做了测试，了解不同因素可能带来的影响。从整体看，主要有以下几点：
+
+- VPC-Native 模式下 Host 网络性能略好于 Pod 网络 (~7%，以 QPS 差异估算，下同)
```
```suggestion
- VPC-Native 模式下 Host 网络性能略好于 Pod 网络（~7%，以 QPS 差异估算，下同）
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+- VPC-Native 模式下 Host 网络性能略好于 Pod 网络 (~7%，以 QPS 差异估算，下同)
+- GCP 的 Ubuntu 系统单纯的读测试中性能略好于 COS (~9%)
+- 使用 Load Balancer 在集群外访问，会略损失性能 (~5%)
+- 多可用区下节点之间延迟增加，会对 TiDB 性能产生一定的影响 (30% ~ 6%，随并发数增加而下降)
```
```suggestion
- 多可用区下节点之间延迟增加，会对 TiDB 性能产生一定的影响（30% ~ 6%，随并发数增加而下降）
```

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,407 @@
+
+Kubernetes 可以允许 Pod 运行在 Host 网络模式下，在 TiDB 实例独占机器没有端口冲突的情况下，此部署方式适用。我们分别对两种网络模式下分别做了 Point Select 测试。
+
+此次测试中，操作系统为 COS 。
```
```suggestion
此次测试中，操作系统为 COS。
```

lilin90 commented:
@cofyc And does this apply to `v3.0` too? If yes, please also add it to `v3.0`. Thanks!

lilin90 commented:
Please add this new file to `TOC.md`.

cofyc commented:
sure, I'll add when this is ready to merge

cofyc commented:
Thanks for reviewing this! Sorry about style issues. 
I've added this file in `TOC.md` and fixed multiple lines issue in https://github.com/pingcap/docs-cn/pull/1803/commits/79e6862c02dd5cdd60258e62c8ffe870e4b5b160.

This applies to `3.0` too. I'll copy to `3.0` after @tennix LGTM this.

lilin90 commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,405 @@
+在单可用区测试中，我们选择了如下型号机器：
+
+| 组件     | 实例类型       | 数量  |
+| ----     | ----------     | ----- |
```
Please use left alignment for all table columns. Thanks!

cofyc replied:
done in https://github.com/pingcap/docs-cn/pull/1803/commits/0cb8281b91876783c0ec894cc323e22456f8c6ed, please check

lilin90 commented:
> Thanks for reviewing this! Sorry about style issues.
> I've added this file in `TOC.md` and fixed multiple lines issue in [79e6862](https://github.com/pingcap/docs-cn/commit/79e6862c02dd5cdd60258e62c8ffe870e4b5b160).
> 
> This applies to `3.0` too. I'll copy to `3.0` after @tennix LGTM this.

It's OK. I guess you must be familiar with the style rules now. ^_^

aylei commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,446 @@
+
+在高并发读测试中，压测端 sysbench 对 CPU 需求较高，需选择多核较高配置型号，避免压测端成为瓶颈。
+
+注：GCP 不同的区域可用机型不同。同时测试中发现磁盘性能表现也存在差异，我们统一在 us-central1 中申请机器测试。
```
Use markdown quote?

cofyc replied:
done in https://github.com/pingcap/docs-cn/pull/1803/commits/a6b9cbc56d0f7a04c0b784b75db9fabcd54e38b3, please check

aylei commented on dev/benchmark/sysbench-in-k8s.md:
````diff
@@ -0,0 +1,446 @@
+  prepare
+```
+
+`<tidb-host`> 为 TiDB 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP、Service 域名、Host IP 以及 Load Balancer IP。（下同）
````
```suggestion
`<tidb-host`> 为 TiDB 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP、Service 域名、Host IP 以及 Load Balancer IP（下同）。
```

lilin90 commented:
@TomShawn Please note that these changes are added to v3.0 in #1809. And use `.png` not `.svg` for images.

TomShawn commented:
> @TomShawn Please note that these changes are added to v3.0 in #1809. And use `.png` not `.svg` for images because [pdflatex does not support `.svg` format](https://github.com/pingcap/docs-cn/pull/1809#issue-312565385).

Sure!

DanielZhangQD commented on dev/benchmark/sysbench-in-k8s.md:
```diff
@@ -0,0 +1,452 @@
+[log]
+level = "error"
+[prepared-plan-cache]
+enabled = true
```
max-procs is not set according to limits.cpu?

cofyc replied:
one tidb instance per node, so no need to set max-procs

DanielZhangQD commented on dev/benchmark/sysbench-in-k8s.md:
````diff
@@ -0,0 +1,452 @@
+内核统一做了如下配置:
+
+```shell
+sysctl net.core.somaxconn=32768
````
The net.core.somaxconn in the container is still 128 even if we set 32768 on the host, will this have any impact on the test?

cofyc replied:
this sysctl parameter determines [backlog of the listen function](https://golang.org/src/net/sock_linux.go), however, we bench performance after all TCP connections are established, so it has no impact on our perf results
